src/python_testing/TC_CHANNEL_5_9.py

```diff
@@ -0,0 +1,150 @@
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --endpoint 1
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#       --PICS src/app/tests/suites/certification/ci-pics-values
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import logging
+
+from chip.clusters import Objects as Clusters
+from chip.interaction_model import Status
+from chip.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main, pics
+from mobly import asserts
+
+
+class TC_CHANNEL_5_9(MatterBaseTest):
+    def desc_TC_CHANNEL_5_9(self) -> str:
+        return "[TC-CHANNEL-5.9] Change Channel Verification (DUT as Server)"
+
+    def steps_TC_CHANNEL_5_9(self) -> list[TestStep]:
+        return [
+            TestStep("0", "Commissioning, already done", is_commissioning=True),
+            TestStep("1", "Read the ChannelList attribute"),
+            TestStep("2", "Read the CurrentChannel attribute"),
+            TestStep("3", "Send ChangeChannel command with a different valid Match string"),
+            TestStep("4", "Verify CurrentChannel updated to the selected channel")
+        ]
+
+    def pics_TC_CHANNEL_5_9(self) -> list[str]:
+        return ["CHANNEL.S"]
+
+    async def read_channel_list(self, endpoint):
+        return await self.read_single_attribute_check_success(
+            endpoint=endpoint,
+            cluster=Clusters.Channel,
+            attribute=Clusters.Channel.Attributes.ChannelList
+        )
+
+    async def read_current_channel(self, endpoint):
+        return await self.read_single_attribute_check_success(
+            endpoint=endpoint,
+            cluster=Clusters.Channel,
+            attribute=Clusters.Channel.Attributes.CurrentChannel
+        )
+
+    async def send_change_channel_command(self, endpoint, match_str):
+        return await self.default_controller.SendCommand(
+            nodeid=self.dut_node_id,
+            endpoint=endpoint,
+            payload=Clusters.Channel.Commands.ChangeChannel(match=match_str)
+        )
+
+    def extract_channel_identifier(self, channel):
+        """Returns a prioritized match string from the channel struct as per spec."""
+        # Priority: NAME > Identifier > AffiliateCallSign > CallSign > Major.Minor
+        if getattr(channel, "name", None):
+            return channel.name
+        if getattr(channel, "identifier", None):
+            return channel.identifier
+        if getattr(channel, "affiliateCallSign", None):
+            return channel.affiliateCallSign
+        if getattr(channel, "callSign", None):
+            return channel.callSign
+        if hasattr(channel, "majorNumber") and hasattr(channel, "minorNumber"):
+            return f"{channel.majorNumber}.{channel.minorNumber}"
```

Comment:
What is the type of `channel`? Do we need to use getattr? Why not things like `if channel.name` or `if cannel.name is not None` ?